hr_attendance_calendar/models/hr_attendance_day.py

```diff
@@ -0,0 +1,370 @@
+
+    # Due hours
+    due_hours = fields.Float('Due hours', compute='_compute_due_hours',
+                             readonly=True, store=True)
+
+    # Attendances
+    attendance_ids = fields.One2many('hr.attendance', 'attendance_day_id',
+                                     'Attendances', readonly=True)
+
+    # Worked
+    worked_hours = fields.Float('Worked hours',
+                                compute='_compute_worked_hours', store=True,
+                                readonly=True)
+    coefficient = fields.Float(readonly=True, string='Coefficient',
+                               help='Worked hours coefficient')
+
+    # Break
+    due_break_min = fields.Float('Minimum break due',
+                                 compute='_compute_due_break')
+    due_break_total = fields.Float('Total break due',
+                                   compute='_compute_due_break')
+    break_ids = fields.One2many('hr.attendance.break',
+                                'attendance_day_id',
+                                'Breaks',
+                                readonly=True)
+    break_max = fields.Float('Longest break',
+                             compute='_compute_break_max',
+                             store=True, )
+    break_total = fields.Float('Total break',
+                               compute='_compute_break_total',
+                               store=True, )
+    rule_id = fields.Many2one('hr.attendance.rules', 'Rules',
+                              compute='_compute_rule_id')
+
+    # Extra hours
+    extra_hours = fields.Float("Extra hours",
+                               compute='_compute_extra_hours',
+                               store=True, )
+
+    ##########################################################################
+    #                             FIELDS METHODS                             #
+    ##########################################################################
+    @api.multi
+    @api.depends('date')
+    def _compute_working_day(self):
+        for att_day in self.filtered('date'):
+            att_day.working_day = fields.Date.from_string(
+                att_day.date).strftime('%A')
+
+    @api.multi
+    @api.depends('working_day')
+    def _compute_name(self):
+        for rd in self.filtered('working_day'):
+            rd.name = rd.working_day + ' ' + rd.date
+
+    @api.multi
+    @api.depends('leave_ids')
+    def _compute_in_leave(self):
+        for att_day in self:
+            if att_day.leave_ids:
+                att_day.in_leave = True
+            if att_day.public_holiday_id:
+                att_day.in_leave = True
+
+    @api.multi
+    @api.depends('cal_att_ids', 'cal_att_ids.due_hours', 'leave_ids.state',
+                 'public_holiday_id')
+    def _compute_due_hours(self):
+        """First search the due hours based on the contract and after remove
+        somme hours if they are vacation"""
+        for att_day in self:
+            due_hours = 0
```

Comment:
For taking the contract hours, you can do it in one line with :  `due_hours = sum(att_day.mapped('cal_att_ids.due_hours'))`